src/main/java/org/commcare/core/graph/c3/DataConfiguration.java

```diff
@@ -373,15 +375,27 @@ private void setColumns(String xID, String yID, SeriesData s) throws GraphExcept
                 xValues.put(barIndex + 1);
                 mBarCount = Math.max(mBarCount, barIndex + 1);
                 if (addBarLabels) {
-                    mBarLabels.put(p.getX());
+                    if (!isRotatedBarGraph) {
+                        currentXLabelStr = p.getX();
+                    }
+                    mBarLabels.put(currentXLabelStr);

```

Comment:
So I'm not making any changes to labels here. The additional code is just for finding the largest labels. 
```
if (!isRotatedBarGraph) {
                        currentXLabelStr = p.getX();
                    }
```
For rotated bar graph, we have rotated axis so not using x axis values to find the largest in this case.

Reply:
sorry don't think I was clear last time. `currentXLabelStr` would be empty when `isRotatedBarGraph` is `true` and would cause `mBarLabels` to add an empty string when earlier it was adding a label. Is that right ?

Reply:
Ohhhh, I see what you're saying now. Thanks!